### ISSUE TITLE
SelfHeal is greedily deleting app folders!

### ISSUE THREAD

**Issue by Anupheaus** [arangodb/arangodb#4395]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3.2

Mode:
- [ ] Cluster
- [x] Single-Server

Storage-Engine:
- [x] mmfiles
- [ ] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [x] docker - official docker library
  - [ ] other:
- [ ] Windows, version:

### Foxx
At startup the SelfHeal function in the foxx manager.js file is greedily trying to delete any folder that is named `app` within the foxx microservice code as well as where the mount points are:
![image](https://user-images.githubusercontent.com/2872785/35276154-503310e8-003a-11e8-9cf3-1b1bf6686748.png)

@pluma pointed out that this is the offending bit of code:
```
  const rootPath = FoxxService.rootPath();
  for (const relPath of fs.listTree(rootPath)) {
    if (!relPath) {
      continue;
    }
    const basename = path.basename(relPath);
    if (basename.toUpperCase() !== 'APP') {
      continue;
    }
    const basePath = path.resolve(rootPath, relPath);
    if (!knownServicePaths.includes(basePath)) {
      modified = true;
      try {
        fs.removeDirectoryRecursive(basePath, true);
        console.debug(`Deleted orphaned service folder ${basePath}`);
      } catch (e) {
        console.warnStack(e, `Failed to delete orphaned service folder ${basePath}`);
      }
    }
```

**Comment by pluma** [comment]
Note to self: this also affects 3.2. IIRC we didn't have that logic in 3.1 so that release should be unaffected. We should do a proper traversal instead of a `listTree` and bail out whenever we have found an `APP` folder (instead of looking inside of that too).

**Comment by OmarAyo** [comment]
Hi @Anupheaus ,

Thanks for reporting this bug

We have released 3.3.4, and it includes a fix for it
I am closing this ticket now. Please feel free to reopen if the problem persists
